cdap-data-fabric/src/main/java/io/cdap/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -158,37 +163,39 @@ public void updateInstance(DatasetId datasetInstanceId, DatasetProperties props)
   @Override
   public Collection<DatasetSpecificationSummary> getInstances(NamespaceId namespaceId)
     throws DatasetManagementException {
-    return clientCache.getUnchecked(namespaceId).getAllInstances();
+    return Retries.callWithRetries(() -> clientCache.getUnchecked(namespaceId).getAllInstances(), retryStrategy);
   }
 
   @Override
   public Collection<DatasetSpecificationSummary> getInstances(NamespaceId namespaceId, Map<String, String> properties)
     throws DatasetManagementException {
-    return clientCache.getUnchecked(namespaceId).getInstances(properties);
+    return Retries.callWithRetries(() -> clientCache.getUnchecked(namespaceId).getInstances(properties), retryStrategy);
   }
 
   @Nullable
   @Override
   public DatasetSpecification getDatasetSpec(DatasetId datasetInstanceId) throws DatasetManagementException {
```

Comment:
Are you planning to add unit tests?

Reply:
Definitely not unit as mocking out all retry mechanism would provide little value, but having a small integration test for exception propagation would be good

Reply:
done